src/views/SearchRedesign/SearchResults.tsx

```diff
@@ -1,79 +1,63 @@
 import { Box, Divider, Flex } from "@chakra-ui/react";
 import { FunctionComponent, useEffect } from "react";
+import { useHistory } from "react-router-dom";
 import { CatalogSearch } from "../../components/CatalogSearch";
 import { PackageList } from "../../components/PackageList";
 import { Page } from "../../components/Page";
-import { Language } from "../../constants/languages";
-import { QUERY_PARAMS } from "../../constants/url";
 import { useCardView } from "../../contexts/CardView";
 import { useCatalogResults } from "../../hooks/useCatalogResults";
-import { useCatalogSearch } from "../../hooks/useCatalogSearch";
-import { useQueryParams } from "../../hooks/useQueryParams";
 import { getSearchPath } from "../../util/url";
 import { PageControls } from "../SearchResults/components/PageControls";
 import { ShowingDetails } from "../SearchResults/components/ShowingDetails";
-import { LIMIT, SearchQueryParam } from "../SearchResults/constants";
-
-const toNum = (val: string) => {
-  const result = parseInt(val);
-
-  if (`${result}` === "NaN") {
-    return 0;
-  }
-
-  return result;
-};
+import { SearchQueryParam } from "../SearchResults/constants";
+import { useSearchState } from "./SearchState";
 
 export const SearchResults: FunctionComponent = () => {
   const { cardView, CardViewControls } = useCardView();
-  const queryParams = useQueryParams();
```

Comment:
A lot of this page-level state has been hoisted to `SearchState.tsx` which implements a react context for components on the page to subscribe to